targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabTests/Public/CppTests/PlayFabCppTests.h.ejs

```diff
@@ -0,0 +1,152 @@
+}
+
+UCLASS()
+class PLAYFABTESTS_API UPlayFabCppTests : public UPlayFabTestCase
+{
+    GENERATED_BODY()
+
+    // Used to end tests.
+    class UPlayFabTestContext* CurrentTestContext = nullptr;
+
+public:
+
+    virtual void GetTests(TArray<UPlayFabTestContext*>& InOutTests) override
+    {
+        InOutTests.ADD_TEST(Invalid Login (CPP), InvalidLogin);
+        InOutTests.ADD_TEST(Login Or Register (CPP), LoginOrRegister);
+        InOutTests.ADD_TEST(Login With Advertising ID (CPP), LoginWithAdvertisingID);
+        InOutTests.ADD_TEST(User Data API (CPP), UserDataAPI);
+        InOutTests.ADD_TEST(Player Statistics API (CPP), PlayerStatisticsAPI);
+        InOutTests.ADD_TEST(LeaderBoard API Client (CPP), LeaderBoardAPIClient);
+        InOutTests.ADD_TEST(LeaderBoard API Server (CPP), LeaderBoardAPIServer);
+        InOutTests.ADD_TEST(Account Info (CPP), AccountInfo);
+        InOutTests.ADD_TEST(Cloud Script (CPP), CloudScript);
+        InOutTests.ADD_TEST(Cloud Script Error (CPP), CloudScriptError);
+        InOutTests.ADD_TEST(Write Event (CPP), WriteEvent);
+        InOutTests.ADD_TEST(Get Entity Token (CPP), GetEntityToken);
+        InOutTests.ADD_TEST(Object API (CPP), ObjectAPI);
+    }
+    
+    PlayFabClientPtr ClientAPI;
+    PlayFabServerPtr ServerAPI;
+    PlayFabAuthenticationPtr AuthenticationAPI;
+    PlayFabDataPtr DataAPI;
+
+    virtual void ClassSetUp() override;
+    virtual void SetUp(UPlayFabTestContext*) override;
+
+    /**
+     * Shared Error callback.
+     * @param InError The PlayFab Error.
+     * @param InCustomData Additional data about this Error.
+     */
+    void OnSharedError(const PlayFab::FPlayFabCppError& InError) const;
+
+    UFUNCTION()
+    void InvalidLogin();
+    void InvalidLogin_Success(const PlayFab::ClientModels::FLoginResult& result);
+    void InvalidLogin_Error(const PlayFab::FPlayFabCppError& InErrorResult); // Expected Result
+
+    UFUNCTION()
+    void LoginOrRegister();
+    void LoginOrRegister_Success(const PlayFab::ClientModels::FLoginResult& result);
+
+    UFUNCTION()
+    void LoginWithAdvertisingID();
+    void LoginWithAdvertisingID_Success(const PlayFab::ClientModels::FLoginResult& result);
+
+    void CheckIfLoggedIn();
+
+    int UserDataAPI_ExpectedValue = -1;
+
+    UFUNCTION()
+    void UserDataAPI(); // GET | UPDATE
+    void GetUserDataAPI_Success(const PlayFab::ClientModels::FGetUserDataResult& result); // GET | UPDATE
+    void UpdateUserDataAPI_Success(const PlayFab::ClientModels::FUpdateUserDataResult& result); // GET | UPDATE
+
+    int PlayerStatistics_ExpectedValue = -1;
+
+    UFUNCTION()
+    void PlayerStatisticsAPI(); // GET | UPDATE
+    void GetPlayerStatisticsAPI_Success(const PlayFab::ClientModels::FGetPlayerStatisticsResult& result); // GET | UPDATE
+    void UpdatePlayerStatisticsAPI_Success(const PlayFab::ClientModels::FUpdatePlayerStatisticsResult& result); // GET | UPDATE
+
+    UFUNCTION()
+    void LeaderBoardAPIClient(); // C | S
```

Comment:
What are these
 // C | S
Comments?

Reply:
Just like above, but with Client & Server LeaderboardAPI. I'll remove this.